### PR TITLE
Fixed `databricks labs ucx repair-run` command to execute correctly

### LINKED ISSUE #787: Cannot repair failed assessment job with `databricks labs ucx repair-run`
Repair run is broken, as it cannot repair a failed workflow:
```
$ databricks labs ucx repair-run --step assessment
12:14:25  INFO [databricks.sdk] Using Databricks CLI authentication
12:14:25  INFO [d.labs.ucx] Repair Running assessment Job
12:14:27 ERROR [d.l.ucx.repair-run] AttributeError: 'NoneType' object has no attribute 'value'
```

### PR DESCRIPTION
**Pull request by prajin-29**
## Changes
Fixing the issue for repair run CLI `databricks labs ucx repair-run` . When a CLI  tries to repair run a job before if updates its response json to either FAILED or SUCCESS it was failing with NoneType exception.

Added a check in `repair_run` inside `install.py` to check the status of the response and wait for 20 seconds to get it updated .

Enhanced the code to repair run already repaired job.  

### Linked issues
closes #787 

Resolves #787 

### Functionality 

- modified the cli command `databricks labs ucx repair-run` which was failing in regression testing

### Tests

- This has been manually tested
- added unit tests `test_repair_run_result_state` in `test_install.py`
- This was tested using integration test
- Screenshot 2024-01-17 at 11 08 42 AM

### CODE REVIEW

File: src/databricks/labs/ucx/install.py [hunk 1]
```diff
@@ -893,13 +895,28 @@ def repair_run(self, workflow):
                 return
             latest_job_run = job_runs[0]
             state = latest_job_run.state
+
+            while not state.result_state and (time.time() - start_time < timeout):
+                logger.info("Waiting for the result_state to update the state")
+                time.sleep(10)
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:901]: This is not unit testable, see how we use retried() decorator in workspace access package (dbsql permissions, secrets acls, etc).

File: src/databricks/labs/ucx/install.py [hunk 2]
```diff
@@ -893,13 +895,28 @@ def repair_run(self, workflow):
                 return
             latest_job_run = job_runs[0]
             state = latest_job_run.state
+
+            while not state.result_state and (time.time() - start_time < timeout):
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:899]: Refactor this into private method and decode with retried

**Review by nfx — CHANGES_REQUESTED** [review]
Rewrite to retried decorator.

File: src/databricks/labs/ucx/install.py (hunk 1 above)
Comment (prajin-29) [reply src/databricks/labs/ucx/install.py]: @nfx .Updated the code with retried logic.

File: src/databricks/labs/ucx/install.py (hunk 2 above)
Comment (prajin-29) [reply src/databricks/labs/ucx/install.py]: @nfx .Refactored the same  with retried logic.

File: src/databricks/labs/ucx/install.py [hunk 3]
```diff
@@ -881,6 +887,15 @@ def latest_job_status(self) -> list[dict]:
...
         return latest_status
 
+    def _get_result_state(self, job_id):
+        job_runs = list(self._ws.jobs.list_runs(job_id=job_id, limit=1))
+        latest_job_run = job_runs[0]
+        if not latest_job_run.state.result_state:
+            logger.info("Waiting for the result_state to update the state")
+            time.sleep(10)
+        job_state = latest_job_run.state.result_state.value
+        return job_state
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:897]:
```suggestion
    def _get_result_state(self, job_id):
        job_runs = list(self._ws.jobs.list_runs(job_id=job_id, limit=1))
        latest_job_run = job_runs[0]
        if not latest_job_run.state.result_state:
            raise AttributeError("no result state in job run")
        job_state = latest_job_run.state.result_state.value
        return job_state
```

Comment (nfx) [reply src/databricks/labs/ucx/install.py:897]: you have `retried(on=[AttributeError]`, but don't throw it anywhere

**Review by nfx — CHANGES_REQUESTED** [review]

**Review by nfx — APPROVED** [review]

**Pull request merged**